src/Aquifer.Public.API/Endpoints/Bibles/Alignments/Greek/Endpoint.cs

```diff
@@ -0,0 +1,275 @@
+    private const string BibleTextWithGreekAlignmentForeignKeysQuery = """
+        SELECT
+            bvw.Id,
+            bvw.WordIdentifier,
+            bvw.[Text] AS EnglishWord,
+            bvw.IsPunctuation,
+            bvwgw.BibleVersionWordGroupId,
+            gntws.GreekSenseId,
+            gntw.GreekWordId
+        FROM BibleVersionWords bvw
+            LEFT JOIN BibleVersionWordGroupWords bvwgw ON bvwgw.BibleVersionWordId = bvw.Id
+            LEFT JOIN BibleVersionWordGroups bvwg ON bvwg.Id = bvwgw.BibleVersionWordGroupId
+            LEFT JOIN NewTestamentAlignments nta ON nta.BibleVersionWordGroupId = BVWG.Id
+            LEFT JOIN GreekNewTestamentWordGroups gntwg ON gntwg.Id = nta.GreekNewTestamentWordGroupId
+            LEFT JOIN GreekNewTestamentWordGroupWords gntwgw ON gntwgw.GreekNewTestamentWordGroupId = GNTWG.Id
+            LEFT JOIN GreekNewTestamentWords gntw ON gntw.Id = gntwgw.GreekNewTestamentWordId
+            LEFT JOIN GreekNewTestamentWordSenses gntws ON gntws.GreekNewTestamentWordId = gntw.Id
+        WHERE
+            bvw.BibleId = @bibleId AND
+            bvw.WordIdentifier BETWEEN @lowerBounds AND @upperBounds AND
+            gntw.Id IS NOT NULL
+        ORDER BY
+            bvw.WordIdentifier,
+            gntw.WordIdentifier,
+            gntws.GreekSenseId
+        """;
+
+    private const string GreekWordsQuery = """
+        SELECT
+            gw.Id,
+            gw.[Text] AS Word,
+            gw.GrammarType,
+            gw.UsageCode,
+            gl.[Text] AS Lemma,
+            sn.[Value] AS StrongsNumber
+        FROM GreekWords gw
+            JOIN GreekLemmas gl ON gw.GreekLemmaId = gl.Id
+            JOIN StrongNumbers sn ON gl.StrongNumberId = sn.Id
+        WHERE gw.Id IN @greekWordIds
+        """;
+
+    private const string GreekSensesQuery = """
+        SELECT
+            gs.Id,
+            gs.DefinitionShort AS [Definition],
+            STRING_AGG(gsg.[Text], '||') AS Glosses
+        FROM GreekSenses gs
+            JOIN GreekSenseGlosses gsg ON gsg.GreekSenseId = gs.Id
+        WHERE gs.Id IN @greekSenseIds
+        GROUP BY
+            gs.Id,
+            gs.DefinitionShort
+        """;
+
+    public override void Configure()
+    {
+        Get("/bibles/{BibleId}/alignments/greek");
+        Options(EndpointHelpers.UnauthenticatedServerCacheInSeconds(EndpointHelpers.OneHourInSeconds));
+        Description(d => d.ProducesProblemFE());
+        Summary(s =>
+        {
+            s.Summary = "Gets a Bible's Greek alignment information.";
+            s.Description =
+                "For a given Bible and book of the Bible, returns the Bible's text along with associated Greek alignment information and Greek sense data for each verse, similar to a reverse interlinear Bible. Data is returned for all verses within the chapter and verse parameters.";
+        });
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
```

Comment:
Overall I think this method is very long, and I would advocate for breaking it up into some smaller private methods (specifically the response building as separately commented below)

Reply:
I split out a bunch of methods, both for data access (which will now be easier to move to a central location in order for the internal API to get at the performance optimizations I made) and for the response creation.  The response creation might be a little harder to understand as a whole now with splitting out the methods but each method is hopefully easier to understand individually.  I also renamed some things for clarity.